Fortnox.NET.Tests/Fortnox.NET.Tests/ContractTemplateTests.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FortnoxNET.Communication;
+using FortnoxNET.Communication.ContractTemplates;
+using FortnoxNET.Models.ContractTemplates;
+using FortnoxNET.Services;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace FortnoxNET.Tests
+{
+    [TestClass]
+    public class ContractTemplateTests : TestBase
+    {
+        [TestMethod]
+        public async Task ItCanGetContractTemplates()
+        {
+            var request = new ContractTemplateListRequest(connectionSettings.AccessToken, connectionSettings.ClientSecret);
+            var response = await ContractTemplateService.GetContractTemplatesAsync(request);
+            
+            Assert.IsNotNull(response);
+            Assert.IsNotNull(response.Data);
+        }
+
+        [TestMethod]
+        public async Task ItCanGetAContractTemplate()
+        {
+            var templates = await GetTemplates();
+
+            if (!templates.Any())
+            {
+                Assert.Inconclusive("No contract templates exist in the system");
+                return;
+            }
+            
+            var request = new ContractTemplateListRequest(connectionSettings.AccessToken, connectionSettings.ClientSecret);
+            var response = await ContractTemplateService.GetContractTemplateAsync(
+                request,
+                templates.First().ContractTemplate
+            );
+            
+            Assert.IsNotNull(response);
+            Assert.IsNotNull(response.Data);
+            Assert.AreEqual(templates.First().ContractTemplate, response.Data.TemplateNumber);
+        }
+
+        // WARNING: Contract Templates have no DELETE endpoint, so running this can add extraneous data to the system
+        // [TestMethod]
+        // public async Task ItCanCreateAContractTemplate()
+        // {
+        //     var template = GenerateTemplate();
+        //     
+        //     var request = new FortnoxApiRequest(connectionSettings.AccessToken, connectionSettings.ClientSecret);
+        //     var response = await ContractTemplateService.CreateContractTemplateAsync(
+        //         request,
+        //         template
+        //     );
+        //     
+        //     Assert.IsNotNull(response);
+        //     Assert.IsNotNull(response.Data);
+        //     Assert.AreEqual(template.TemplateName, response.Data.TemplateName);
+        // }
+
+        [TestMethod]
+        public async Task ItCanUpdateAContractTemplate()
+        {
+            var templates = await GetTemplates();
+
+            if (!templates.Any())
+            {
+                Assert.Inconclusive("No contract templates exist in the system");
+                return;
```

Comment:
no need for the return statement, Assert will terminate

Reply:
resolved